web3.js/src/connection.ts

```diff
@@ -4283,6 +4412,26 @@ export class Connection {
     rawConfig?: GetBlockConfig,
   ): Promise<BlockResponse | null>;
 
+  /**
+   * @deprecated Instead, call `getBlock` using a `GetVersionedBlockConfig` by
+   * setting the `maxSupportedTransactionVersion` property.
+   */
+  // eslint-disable-next-line no-dupe-class-members
+  async getBlock(
+    slot: number,
+    rawConfig: GetBlockConfig & {transactionDetails: 'accounts'},
+  ): Promise<AccountsModeBlockResponse | null>;
```

Comment:
This is how we should have done `getParsedBlock` back in the day. This method signature overload tells Typescript what shape the return value should have in the case that this method is called with exactly `accounts` as the `transactionDetails` property. We could have done the same with the `encoding` property, obviating the need for the `getParsedBlock` method altogether.

cc/ @jordansexton, @ngundotra, @lwus, @2501babe, @timhagn, @catalinred for the TypeScript trivia.